elderly_assistant/services/speech.py

```
@@ -70,34 +70,105 @@ def _init_engines(self):
+            preferred_voice = self._select_mbrola_voice(eng)
+        except Exception as e:
+            self.logger.warning(f"pyttsx3 语音选择失败（使用默认语音）: {e}")
+            preferred_voice = None
+
+        # CodeRabbit 修复：在发布引擎前校验 volume/rate/voice 的最终生效值。
+        # 部分后端 setProperty 返回 True 但并未真正生效（或发音后端延迟失败），
+        # 这里用 runAndWait 做一次空播报触达后端，并回读校验；音量/语速校验
+        # 失败则丢弃该引擎，语音校验（优选语音"假成功"）失败则恢复保存的默认语音。
+        _valid = True
+        try:
+            # 回读校验 volume / rate 是否生效（允许一定误差）
```

Ruff (0.16.1)

[warning] Comment contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF003)